uservice_ccutter/plugins/substitute.py

```diff
@@ -15,15 +14,12 @@ def substitute(templatetype, auth, inputdict):
     GitHub.  That does imply that any template whatsoever must have at least
     one field that's guaranteed to be present so that we can use its hook to
     drive creation of the github_* fields."""
-    ctt = "." + templatetype.lower().replace("-", "_")
-    ctr = ".".join(__name__.split(".")[:-1])
-    importlib.import_module(ctt, package=ctr)
-    modname = ctr + ctt
-    symbols = [x for x in sys.modules[modname].__dict__ if x[0] != "_"]
+    module = load_plugin(templatetype)
+    symbols = [x for x in module.__dict__ if x[0] != "_"
+               and x != "finalize_"]
```

Comment:
I foresee some maintainability issues by putting knowledge of `finalize_` here into `substitute`. Maybe it'd be better to filter out any object with a name that also *ends* in an underscore, `x[-1] != '_'`.

Then we could add more hooks like `finalize_` and there wouldn't be any need to update the substitution discovery code.